src/Concordium/Client/Runner.hs

```diff
@@ -500,6 +506,24 @@ loadAccountImportFile format file name = do
      accCfg <- decodeGenesisFormattedAccountExport contents name pwd `withLogFatalIO` ("cannot import account: " ++)
      return [accCfg]
 
+-- Read a transaction memo as either
+-- - a CBOR encoded string,
+-- - CBOR encoded json (from a file), or
+-- - raw bytes (from a file). 
+readMemoAsBytestring :: MemoInput -> Backend -> IO ByteString
+readMemoAsBytestring memo backend = do
```

Comment:
I think the following function should either take the protocol version as a parameter so it does not have to use the backend itself, or it should be changed so that it lives in the `ClientMonad IO`.

The reason for the latter is that as it is now whenever this function is used it creates a connection to the node, and then drops it. The connection is later re-established to send the transaction.

Since this function also checks whether a memo is allowed it should probably be renamed to reflect this, something along the lines of `checkAndGetMemo`.